csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -209,14 +268,13 @@ private void WriteMessage(TextWriter writer, IMessage message)
                 }
             }
             writer.Write("{ ");
-            bool writtenFields = WriteMessageFields(writer, message, false);
-            writer.Write(writtenFields ? " }" : "}");
+            bool nothingWritten = WriteMessageFields(writer, message, true);
+            writer.Write(nothingWritten ? "}" : " }");
         }
 
-        private bool WriteMessageFields(TextWriter writer, IMessage message, bool assumeFirstFieldWritten)
+        private bool WriteMessageFields(TextWriter writer, IMessage message, bool firstField)
```

Comment:
Ah OK, now I understand why you did this in the first place. It was not clear to me that you did all this bool-flipping for this purpose. If you find it easier to read that way, I will adapt my changes to this logic and won't reverse the meaning.